usaspending_api/common/helpers/orm_helpers.py

```diff
@@ -51,9 +84,50 @@ def generate_raw_quoted_query(queryset):
 
 
 def generate_where_clause(queryset):
-    """
-    Returns the SQL and params from a queryset all ready to be plugged into an
-    extra method.
-    """
+    """Returns the SQL and params from a queryset all ready to be plugged into an extra method."""
     compiler = queryset.query.get_compiler(get_connection().alias)
     return queryset.query.where.as_sql(compiler, compiler.connection)
+
+
+def obtain_view_from_award_group(type_list):
+    if not type_list:
+        raise AwardGroupsException("Invalid award type list: No types provided.")
+
+    type_set = set(type_list)
+    for category, values in CATEGORY_TO_MODEL.items():
+        if type_set <= values["types"]:
+            return values["model"]
+    else:
+        raise AwardGroupsException("Invalid award type list: Types cross multiple categories.")
```

Comment:
Technically couldn't reaching here also mean that you picked a type not in any category?